cla_backend/apps/call_centre/tests/test_forms.py

```diff
@@ -482,6 +480,28 @@ def _test(case, datetime, error_msg):
 
         _test(case, self._strftime(mon), u"Specify a date within working hours.")
 
+    @mock.patch("call_centre.forms.timezone.now")
+    def test_valid_datetime(self, mocked_now):
+        now_utc = datetime.datetime(2015, 3, 13, 10, 15).replace(tzinfo=timezone.utc)
+        mocked_now.return_value = now_utc
+
+        case = make_recipe("legalaid.case")
+
+        def _test(case, datetime):
+            data = self.get_default_data()
+            data["datetime"] = datetime
+            form = self.FORM(case=case, data=data)
+
+            self.assertTrue(form.is_valid())
+
+        # datetime in the future
+        _test(case, self._strftime(now_utc + datetime.timedelta(minutes=10)))
+        _test(case, self._strftime(now_utc + datetime.timedelta(minutes=30)))
+        _test(case, self._strftime(now_utc + datetime.timedelta(minutes=180)))
+
+        # datetime in the current 30 minute window
+        _test(case, self._strftime(now_utc - datetime.timedelta(minutes=14)))
```

Comment:
Is line 498 not testing the same thing as line 503?

Reply:
Line 503 is in the past, subtracted timedelta. I did wonder if that was easy to miss, and it is, so I'll probably change it to being `+ datetime.timedelta(minutes=-14)` , which I think does the same thing but is easier to spot

Reply:
Oh no sorry Michael, I meant aren't both tests testing for the datetime within the same callback window? For example, if the time was 10.15am, wouldn't -14minutes and +10minutes from 10.15am still be within the same callback window?

Reply:
Ah yes I guess so, good catch! Do you think I should delete the 10 minute one?

Reply:
Yeah I think with the comments it would make sense to delete the 10min one. To nitpick, could we change -14mins to -10mins just so it's more obvious?

Reply:
I don't see what that would achieve

Reply:
Yeah I realised it doesn't matter. What the test is doing is more important + the comment is already there

Reply:
Actually, I'm going to leave in both the past-but-same-window and future-but-same-window, as I think having them both is slightly valuable against regression, but it was a very good catch 👍